modules/highgui/src/window_framebuffer.cpp

```diff
@@ -0,0 +1,783 @@
+            return -1;
+        }
+
+        xvfb_len_header = C32INT(&(xwd_header->header_size));
+        xvfb_len_colors = sizeof(XWDColor) * C32INT(&(xwd_header->ncolors));
+        xvfb_len_pixmap = C32INT(&(xwd_header->bytes_per_line)) *
+        C32INT(&(xwd_header->pixmap_height));
+
+        munmap(xwd_header, sizeof(XWDFileHeader));
+
+        fbScreenSize = xvfb_len_header + xvfb_len_colors + xvfb_len_pixmap;
+        xwd_header = (XWDFileHeader*)
+        mmap(NULL, fbScreenSize, PROT_READ | PROT_WRITE, MAP_SHARED, fb_fd, 0);
+
+        fbPointer = (unsigned char*)xwd_header;
+        fbPointer_dist = xvfb_len_header + xvfb_len_colors;
+
+        return fb_fd;
+    }
+
+    fb_var_screeninfo &FramebufferBackend::getVarInfo()
+    {
+        return varInfo;
+    }
+
+    fb_fix_screeninfo &FramebufferBackend::getFixInfo()
+    {
+        return fixInfo;
+    }
+
+    int FramebufferBackend::getFramebuffrerID()
+    {
+        return fbID;
+    }
+
+    int FramebufferBackend::getFBWidth()
+    {
+        return fbWidth;
+    }
+
+    int FramebufferBackend::getFBHeight()
+    {
+        return fbHeight;
+    }
+
+    int FramebufferBackend::getFBXOffset()
+    {
+        return fbXOffset;
+    }
+
+    int FramebufferBackend::getFBYOffset()
+    {
+        return fbYOffset;
+    }
+
+    int FramebufferBackend::getFBBitsPerPixel()
+    {
+        return fbBitsPerPixel;
+    }
+
+    int FramebufferBackend::getFBLineLength()
+    {
+        return fbLineLength;
+    }
+
+    unsigned char* FramebufferBackend::getFBPointer()
+    {
+        return fbPointer + fbPointer_dist;
+    }
+
+    Mat& FramebufferBackend::getBackgroundBuff()
+    {
+        return backgroundBuff;
+    }
+
+    OpenCVFBMode FramebufferBackend::getMode()
+    {
+        return mode;
+    }
+
+    FramebufferBackend::FramebufferBackend():mode(FB_MODE_FB), fbPointer_dist(0)
+    {
+        CV_LOG_INFO(NULL, "UI: FramebufferWindow::FramebufferBackend()");
+
+        std::string fbModeStr = getFBMode();
+
+        if (fbModeStr == "EMU")
+        {
+            mode = FB_MODE_EMU;
+            CV_LOG_WARNING(NULL, "UI: FramebufferWindow is trying to use EMU mode");
+        }
+        if (fbModeStr == "FB")
+        {
+            mode = FB_MODE_FB;
+            CV_LOG_WARNING(NULL, "UI: FramebufferWindow is trying to use FB mode");
+        }
+        if (fbModeStr == "XVFB")
+        {
+            mode = FB_MODE_XVFB;
+            CV_LOG_WARNING(NULL, "UI: FramebufferWindow is trying to use XVFB mode");
+        }
+
+        fbID = -1;
+        if (mode == FB_MODE_FB)
+        {
+            fbID = fbOpenAndGetInfo();
+        }
+        if (mode == FB_MODE_XVFB)
+        {
+            fbID = XvfbOpenAndGetInfo();
+        }
+
+        CV_LOG_INFO(NULL, "UI: FramebufferWindow::fbID " << fbID);
+
+        if (fbID == -1)
+        {
+            mode = FB_MODE_EMU;
+            fbWidth = 1024;
+            fbHeight = 768;
```

Comment:
Why not just VGA? It's more common in cv.

Reply:
changed the resolution to 640x480